File: 3ahwii/2025-11_brueche/Bruch.cs
class Bruch
{
    // jetzt kommen die sog. "Attribute" der Klasse oder "Felder"
    private int zaehler;
    private int nenner;

    public Bruch(string bruchtext)
    {
        string[] teile = bruchtext.Split('/');
        this.zaehler = int.Parse(teile[0]);
        this.nenner = int.Parse(teile[1]);
    }
    public Bruch addiere(Bruch b)
    {
        return null;
    }
    public string toString()
    {
        return $"ich bin ein bruch: {this.zaehler}/{this.nenner}";
        // JS: return `ich bin ein bruch: ${this.zaehler}/${this.nenner}`;
    }   
}

File: 3ahwii/2026-09-08_intro/bruch.ts
export class Bruch {
  private zähler: number;
  private nenner: number;

  constructor(zähler: number, nenner: number) {
    this.zähler = zähler;
    this.nenner = nenner;
  }

  addiere(other: Bruch): Bruch {
    return new Bruch(
      this.zähler * other.nenner + other.zähler * this.nenner,
      this.nenner * other.nenner,
    );
  }

  toString(): string {
    return `${this.zähler}/${this.nenner}`;
  }
}
